### PR TITLE
Kubernetes License check for Kubernetes Proxies

### PR DESCRIPTION
**Pull request by tigrato**
This PR ensures that a given Teleport root cluster is licensed for Kubernetes access when forwarding credentials to leaf clusters.

Previously, the root Kube proxy would call auth server to generate a new cert-key pair with the user identity and during the call Auth ensured that it was licensed for Kubernetes usage. Given the recent developments for #22533, the call mentioned was removed and it's now possible for a root enterprise cluster to forward requests to a OSS leaf cluster without validating its license.

Auth server already enforces Kubernetes license for Kubernetes service when it tries to heartbeat the cluster. If the cluster is not properly licensed, it's not possible to register kubernetes clusters.

Part of #22533

### CODE REVIEW

**Review by espadolini — COMMENTED** [review]
You can just use `(*TeleportProcess).getClusterFeatures` instead of running your own periodic check; it gets updated whenever any service in the process (re)connects to the auth. Up to you if you think you should check it every time (by passing some closure) or only whenever we initialize the forwarder for the first time.

**Review by zmb3 — APPROVED** [review]

File: lib/kube/proxy/forwarder.go [hunk 1]
```diff
@@ -498,6 +509,12 @@ const accessDeniedMsg = "[00] access denied"
 
 // authenticate function authenticates request
 func (f *Forwarder) authenticate(req *http.Request) (*authContext, error) {
+	// If the cluster is not licensed for Kubernetes, return an error to the client.
+	if !f.cfg.ClusterFeatures().Kubernetes {
+		// If the cluster is not licensed for Kubernetes, return an error to the client
+		// but in a way that the code is translated to 500 error code.
```
Comment (Tener) [inline lib/kube/proxy/forwarder.go:515]: If this isn't too much hassle, it might be worth testing that the error is truly translated to a 500 error code.

**Review by Tener — APPROVED** [review]

File: lib/kube/proxy/forwarder.go (hunk 1 above)
Comment (tigrato) [reply lib/kube/proxy/forwarder.go:515]: Added in 9a249562a6963d76981bdeb3e9939f04378c308f

Comment (espadolini) [reply lib/kube/proxy/forwarder.go:515]: Should this be a 500 or a 403 (Forbidden)?

Comment (espadolini) [reply lib/kube/proxy/forwarder.go:515]: We seem to return `trace.AccessDenied` (which becomes a 403 in http) whenever some license feature check fails, which seems appropriate here. Is there a kube-specific reason why it should not be returned as an access denied?

Comment (tigrato) [reply lib/kube/proxy/forwarder.go]: I preferred 500 to distinguish from cases where the user does not have access but reverted to Forbidden

**Review by espadolini — APPROVED** [review]

**Pull request merged**